pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -87,6 +87,9 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		return err
 	}
 
+	// maintain the set's revision history limit
+	defer ssc.truncateHistory(set, pods, revisions, currentRevision, updateRevision)
```

Comment:
Maybe this should be above the call to `getStatefulSetRevisions`? Revisions would be created inside that call, so it seems like we should make sure we check if the history should be truncated after every call.

Reply:
[This commit](https://github.com/kubernetes/kubernetes/pull/86097/commits/a00b9905498f2695bab35f891739fe02116eb2dd) moves `defer ssc.truncateHistory()` further up above the call to `getStatefulSetRevisions`.

However, after a second look, I've decided to keep `truncateHistory` after `getStatefulSetRevisions` because `truncateHistory` only cleans up **non-live** revisions. The revision that gets created inside `getStatefulSetRevisions` is considered as [**live** revision](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/statefulset/stateful_set_control.go#L154) and will not be cleaned up anyway. Hence, to keep the code clean, the commit (https://github.com/kubernetes/kubernetes/pull/86097/commits/a00b9905498f2695bab35f891739fe02116eb2dd) was reverted.